### ISSUE TITLE
latex formula embeded

### ISSUE THREAD

**Issue by GuangchuangYu** [rich-iannone/DiagrammeR#91]
Is it possible to label the text with latex formula?

**Comment by rich-iannone** [comment]
@GuangchuangYu it's not currently possible but I think it's a good feature to have! I'll work toward implementing this soon.

**Comment by fdabl** [comment]
yes, this is the killer feature for me! any progress update? great software, by the way :)

**Comment by rdatasculptor** [comment]
maybe there is a possibility to add katexR features to DiagrammeR? Just a thought.
http://www.buildingwidgets.com/blog/2015/2/5/week-05-katex-in-r

**Comment by rich-iannone** [comment]
@timelyportfolio , I'd forgotten about this awhile back but it looks like the perfect way to get equations and such into node labels. Do you think it could be feasible to implement? Perhaps the node attribute katex could be used and parsed by `render_graph()`.

**Comment by rich-iannone** [comment]
And thank you @fdabl for bringing this to the fore and @rdatasculptor for the recommendation!

**Comment by timelyportfolio** [comment]
I'll look into it this afternoon.

**Comment by timelyportfolio** [comment]
Unfortunately, this is much trickier than I hoped.

**Comment by rich-iannone** [comment]
@timelyportfolio thanks for trying! :)

**Comment by timelyportfolio** [comment]
I discovered this repo https://github.com/ichuang/svg_mathjax2 today, and it might just woork for LaTeX in `grViz`.  I have worked up some quick code in a forked branch.  Here is a quick example.

```
#devtools::install_github("timelyportfolio/DiagrammeR@feature/svg-mathjax"
library(DiagrammeR)

add_mathjax(
  grViz(
'
digraph G {
  A [label="$\\\\cos (2\\\\theta) = \\\\cos^2 \\\\theta - \\\\sin^2 \\\\theta$"];
  A -> B;
}
'
  )
)
```

![image](https://cloud.githubusercontent.com/assets/837910/18020286/48a7bc90-6ba6-11e6-82ee-081889ead8a3.png)

**Comment by rich-iannone** [comment]
@timelyportfolio This is fantastic! I've been looking for this exact functionality. I'm definitely going to work this into DiagrammeR. Thanks!

**Comment by rich-iannone** [comment]
@timelyportfolio forgot to ask in my excitement: would you be able to make a PR for this?

**Comment by timelyportfolio** [comment]
Let's test a little bit before PR time.  Happy to submit PR after a little testing.  Also, please be sure to see the deficiencies that I added in the discussion above.

**Comment by rich-iannone** [comment]
@timelyportfolio yes, testing is always a good thing. Noted these deficiencies and it does seem like there might be more little issues.

**Comment by timelyportfolio** [comment]
Any follow up?

**Comment by rich-iannone** [comment]
@timelyportfolio would you be able to submit a PR for this? Looks pretty solid and it would be a great addition.

**Comment by MauroCE** [comment]
Any updates?

**Comment by josefwilzen** [comment]
Any updates on this feature?

**Comment by chuckleong21** [comment]
> I discovered this repo https://github.com/ichuang/svg_mathjax2 today, and it might just work for LaTeX in `grViz`. I have worked up some quick code in a forked branch. Here is a quick example.
> 
> ```
> #devtools::install_github("timelyportfolio/DiagrammeR@feature/svg-mathjax"
> library(DiagrammeR)
> 
> add_mathjax(
>   grViz(
> '
> digraph G {
>   A [label="$\\\\cos (2\\\\theta) = \\\\cos^2 \\\\theta - \\\\sin^2 \\\\theta$"];
>   A -> B;
> }
> '
>   )
> )
> ```
> 
> ![image](https://cloud.githubusercontent.com/assets/837910/18020286/48a7bc90-6ba6-11e6-82ee-081889ead8a3.png)
> 
> ### known deficiencies
> * as far as I can tell, the shapes are not resized for the rendered LaTeX content, so we need to hope the rendered content is close to the same size as the expression
> * `rmarkdown` by default adds MathJax, so I have added an option `include_mathjax` to turn off the duplicate addition of MathJax, since it is a very heavy dependency
> * have not tried with multiple `grViz`, but I used `htmlwidgets::onStaticRenderComplete()` to wait for all widgets to be rendered before running
> * uses a sloppy `setTimeout` with 3000ms delay to make sure we have MathJax and htmlwidgets are completely rendered
> * ... probably more but these are the ones I have already discovered

As of version 1.0.11, this feature is available through the proposal by @timelyportfolio above. Here's my test case: 
![r2_derivation](https://github.com/user-attachments/assets/e09d78e8-fa68-4b7c-99bc-59107346a104)

A couple of things to note, though:
- Plain text right after the latex block wrapped by `$` makes the latex block not being rendered. However, `\n` could be used as a divider for these two: `"$x^2\nis unknown$"`
- Latex blocks are not affected by attributes such as `fontsize` in dot graph. They are entirely controlled by related keywords like `\large{}``\textrm{}`
